### ISSUE TITLE
Relation in Mutations do not return correct data since alpha40

### ISSUE THREAD

**Issue by lumannnn** [hasura/graphql-engine#1794]
Hey there! :)

First of all, you guys do amazing work and Hasura is a really great product! ❤️🚀 

We have an issue in a ``delete mutation`` which might be related to https://github.com/hasura/graphql-engine/issues/1763.

We try to delete a row of a table and in the response of the mutation, we also want to return data of the relation to which it was related before we deleted it.

## Example

```
Table foo
----------
id bigint primary key
name string
bar_id bigint  (has an object relation to "bar" - note: we do not yet use foreign key constraints)

Table bar
---------
id bigint primary key
name string
```

```graphql
mutation Delete($id: bigint!) {
  delete_foo(where: {id: {_eq: $id}}) {
    affected_rows
    returning {
      id
      name
      bar_id
      bar {
        id
      }
    }
  }
}
```

## Problem

If we specify the response of the query to also return data of the ``bar``-relation we always get

```json
{
  "affected_rows": 0,
  "returning": []
}
```

However, the ``foo``-row(s) are deleted nonetheless! Only the response is wrong.
If we do not want the query to return data of the relation (i.e. omitting ``bar { id }``, then we do get proper data for the other values (``foo.id, foo.name, foo.bar_id``) in the response.

```json
{
  "affected_rows": 1,
  "returning": [
    {
      "id": 1,
      "name": "something",
      "bar_id": 3
    }
  ]
}
```

All tables in our system do have primary keys - all primary keys are ``bigint``s.

This worked in version ``alpha37`` (not sure about the other versions - we haven't upgraded since then).

Can anyone replicate this behaviour or are we doing anything wrong on our side?

**Comment by 0x777** [comment]
Hi, this is related to #1763. The change in `alpha40` to fix #1576 seems to be causing more issues than what it attempts to fix. We'll have a discussion tomorrow and release a fix as soon as possible.

Prior to `alpha40`, this mutation (I added `foos` to the selection set of `bar`) would also return the deleted foo in `foos` of `bar`. That's the corner case `alpha40` attempts to fix.

```graphql
mutation Delete($id: bigint!) {
  delete_foo(where: {id: {_eq: $id}}) {
    affected_rows
    returning {
      id
      name
      bar_id
      bar {
        id
        foos { id }
      }
    }
  }
}
```

**Comment by lumannnn** [comment]
Thanks for the fast response!

Yep, I see. Totally would make sense if in that case the deleted ``foo`` would not be included in the ``bar.foos`` anymore.

Looking forward to the fix. Let me know if I can help you with testing or providing further information about our setup.

Thanks and keep up the good work! 🙂👍
